argo/common/run_completion_event_unit_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build unit
+
+package common
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Context("RunCompletionEvent.Validate", func() {
+
+	validate, _ = InitialiseValidation()
+
+	namespaceName := NamespacedName{
+		Name:      "Name",
+		Namespace: "Namespace",
+	}
+
+	baseEvent := RunCompletionEvent{
+		Status:               "succeeded",
+		PipelineName:         namespaceName,
+		RunConfigurationName: &namespaceName,
+		RunName:              &namespaceName,
+		RunId:                "RunId",
+		Provider:             "Provider",
+	}
+
+	It("is valid when RunConfigurationName is missing and RunName is provided", func() {
+		invalidEvent := baseEvent
+		invalidEvent.RunConfigurationName = nil
+		Expect(invalidEvent.Validate(validate)).To(Succeed())
```

Comment:
because reading this it says an invalid event is valid 🤯

Reply:
yup needs to change